### ISSUE TITLE
Initial login with current quelpa version

### ISSUE THREAD

**Issue by mullikine** [alphapapa/ement.el#21]
https://mullikine.github.io/posts/feedback-for-alphapapa-ement-el-matrix-client-for-emacs/

The initial login is currently broken, but if the token already exists, there is no login problem.

The blog article contains a video demonstration.

**Comment by alphapapa** [comment]
I don't understand.  At the end of that, you `load-library` on `matrix-client`, which is not part of this package.

Please, just describe the error in words and post it here.  Please don't ask me to go watch a video about something that went wrong unless it can only be shown in video.

**Comment by mullikine** [comment]
My mistake. I copied the wrong segment of code into the description.

The issue can be described as so:

- I clear the token
- I load with quelpa, using the following code:

```
;; Install and load `quelpa-use-package'.
(package-install 'quelpa-use-package)
(require 'quelpa-use-package)

;; Install `plz' HTTP library (not on MELPA yet).
(use-package plz
  :quelpa (plz :fetcher github :repo "alphapapa/plz.el"))

;; Install Ement.
(use-package ement
  :quelpa (ement :fetcher github :repo "alphapapa/ement.el"))
```

- I then `ement-connect`, enter my details and then an error appears

This is the backtrace:
```
Debugger entered--Lisp error: (wrong-type-argument listp {)                                                                                                     
  cdr({)                                                                                                                                                        
  (car (cdr (read (current-buffer))))                                                                                                                           
  (save-excursion (beginning-of-defun) (car (cdr (read (current-buffer)))))                                                                                     
  (save-current-buffer (set-buffer (or byte-compile-current-buffer (current-buffer))) (save-excursion (beginning-of-defun) (car (cdr (read (current-buffer))))))
  (let* ((fn-name (save-current-buffer (set-buffer (or byte-compile-current-buffer (current-buffer))) (save-excursion (beginning-of-defun) (car (cdr (read ...))
  (closure (t) (&rest args) "Display a debug warning showing the runtime value ..." (let* ((fn-name (save-current-buffer (set-buffer (or byte-compile-current-bu
  (ement-debug "Updating progress reporter to" value)                                                                                                           
  (progn (if --cl-rest-- (signal 'wrong-number-of-arguments (list nil (+ 1 (length --cl-rest--))))) (ement-debug "Updating progress reporter to" value) (progres
  (let* ((value (if --cl-rest-- (pop --cl-rest--) (cl-incf g3154)))) (progn (if --cl-rest-- (signal 'wrong-number-of-arguments (list nil (+ 1 (length --cl-rest-
  ement-progress-update()                                                                                                                                       
  (progn (ement-progress-update))                                                                                                                               
  (if (ement--sync-messages-p session) (progn (ement-progress-update)))      
```

I then load `ement` manually from the latest commit.

```
(add-to-list 'load-path (concat emacsdir "/manual-packages/ement.el"))
(require 'ement)
(load-library "ement")
```

Then try to connect and the error no longer occurs.

**Comment by alphapapa** [comment]
From studying the backtrace, it appears that you may be doing something unusual in your Emacs config.  The error comes from an expansion of the `ement-debug` macro, and it appears that it's not being expanded until runtime, which would suggest that you're loading the non-byte-compiled version of the file.  But there's no need for you to do that, because you installed the package with Quelpa, which byte-compiles the files when it installs the package.

Maybe just try omitting the `load-library` call, which is redundant, because you already `require`d it.

In other words, if you just follow the instructions as written, rather than adding to them, it will work.  ;)
